app/logic/events.py

```diff
@@ -325,16 +323,27 @@ def preprocessEventData(eventData):
         eventData['timeEnd'] = format24HourTime(eventData['timeEnd'])
 
     ## Get the facilitator objects from the list or from the event if there is a problem
-    try:
-        if type(eventData) == MultiDict and type(eventData['facilitators']) is not list:
-            eventData['facilitators'] = eventData.getlist('facilitators')
-        eventData['facilitators'] = [User.get_by_id(f) for f in eventData['facilitators']]
-    except Exception as e:
-        event = eventData.get('id', -1)
-        eventData['facilitators'] = list(User.select().join(EventFacilitator).where(EventFacilitator.event == event))
+    if 'facilitators' in eventData:
+        eventData['facilitators'] = getFacilitatorsFromList(eventData['facilitators'])
 
     return eventData
 
+def getFacilitatorsFromList(facilitatorList):
+    """
+    This function takes in a list of usernames or a string with the usernames of
+        facilitators separated by commas (,) and returns a list of facilitator
+        objects that match the usernames in the list
+    facilitatorList: expected to be a list with facilitator usernames as strings
+                        or a string of usernames separated by commas (,)
+    return: list of facilitator objects matching the usernames in facilitatorList
+    """
+    if facilitatorList and type(facilitatorList) == str:
+        facilitatorList = facilitatorList.split(',')
+    finalFacilitatorList = []
+    # for each facilitator in facilitatorList, if it is not a blank string of null
+    # add a user object matching said facilitator to finalFacilitatorList
+    finalFacilitatorList = [User.get_by_id(facilitator) for facilitator in facilitatorList if facilitator]
```

Comment:
Is there a case where we need that `if facilitator`? We probably want to have an error if there is weird data being passed in